iiod-responder.c

```diff
@@ -414,33 +421,47 @@ bool iiod_io_has_response(struct iiod_io *io)
 {
 	uint64_t timeout_us = io->timeout_ms * 1000;
 
-	return io->r_done ||
-		(read_counter_us() - io->w_io.start_time > timeout_us);
+	if (io->r_done)
+		return true;
+
+	if (!io->timeout_ms)
+		return false;
+
+	timeout_us = io->timeout_ms * 1000;
+
+	return read_counter_us() - io->w_io.start_time > timeout_us;
+}
+
+static int iiod_io_cond_wait(struct iiod_io *io)
```

Comment:
Can we make `struct iiod_io *io() `const`? Or the calls to `iio_cond_wait()` will WARN?

Reply:
Yes it can be const, I'll change it. It will be fine with `io->cond` and `io->lock` because those are pointers, so making `io` const will make these immutable pointers and not pointers to const data.